### ISSUE TITLE
Improve database handling

### ISSUE THREAD

**Issue by Hyodar** [NethermindEth/near-sffl#210]
At the very beginning of development, we were using an embedded KV database - it was performant, simple and good enough for our uses. After we started getting on with more complex queries, we decided to move on to SQLite, since it's a good alternative for an SQL database that doesn't require much of a setup and still holds pretty good performance.
Still, now we're seeing issues such as #209 which brings the question of which tooling we should use. With `gorm`, we're flexible enough to switch to some other SQL database without much of an issue.

Making this issue so we can discuss the alternatives and pros and cons here.

**Comment by emlautarom1** [comment]
I agree on keeping an SQL interface rather than using plain KV since, as you mention, this allows us to perform complex queries if needed. At the same time - if possible - I would prefer to not spin up an additional server (ex. Postgres) since it adds additional maintenance cost down the road. A couple of questions right now are:
- Do we see performance degradation by limiting SQLite to a single connection?
- If there is, does it have anything to do with the ORM we're using?
- If we want concurrent access, then WAL (https://www.sqlite.org/wal.html) sounds like a well supported feature since it's been available since 2010.

**Comment by Hyodar** [comment]
Let's keep this open for a while. I don't think our use-case should be problematic with a single connection, but if it comes to that this discussion should be brought.
